Fix disconnect freeze

RastislavKamenickyOnsemi opened the pull request:
Fixed freeze when connection was not available, fixed freeze when connection was abruptly terminated, updated couchbaselite example (on which it was discovered/tested).

RastislavKamenickyOnsemi committed:
show debug logs in console

RastislavKamenickyOnsemi committed:
updated sgcouchbaselite example to use proper log macros

RastislavKamenickyOnsemi committed:
fixed freeze when connection failed to establish

RastislavKamenickyOnsemi committed:
speedup the termination to not wait for too long

luayalshawionsemi commented:
please explain how this issue can be reproduced and is it caused by the previous pull-request merge?

RastislavKamenickyOnsemi commented:
> please explain how this issue can be reproduced and is it caused by the previous pull-request merge?

In couchbaselite example:
- start the example (without the local repo it is trying to connect), it will fail to connect and then will hung at exit. This is caused INDIRECTLY by previous pull request - https://github.com/stratadeveloperstudio/couchbase-lite-cpp/pull/16 because the Puller for which Replicator is waiting to terminate, will remain stuck. I added a check in Puller/Pusher to make sure they stop if they get disconnected.

As part of Strata Developer Studio:
- start SDS, login, disconnect internet, quit SDS. It will hung at trying to close socket. I did not found any setting related to socket timers for that operation, so I added additional timer (500msec) which will close the socket by force if it does not terminates by then.

Reproduced at both Windows and MacOS.

RastislavKamenickyOnsemi committed:
a more graceful exit if connection is working properly

RastislavKamenickyOnsemi committed:
removed obsolete SHOW_DATABASE_MESSAGES define

luayalshawionsemi commented:
> > please explain how this issue can be reproduced and is it caused by the previous pull-request merge?
> 
> In couchbaselite example:
> 
> * start the example (without the local repo it is trying to connect), it will fail to connect and then will hung at exit. This is caused INDIRECTLY by previous pull request - #16 because the Puller for which Replicator is waiting to terminate, will remain stuck. I added a check in Puller/Pusher to make sure they stop if they get disconnected.
> 
> As part of Strata Developer Studio:
> 
> * start SDS, login, disconnect internet, quit SDS. It will hung at trying to close socket. I did not found any setting related to socket timers for that operation, so I added additional timer (500msec) which will close the socket by force if it does not terminates by then.
> 
> Reproduced at both Windows and MacOS.

Thanks for the explanation.
